### PR TITLE
Complete smoke tests

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T335160

Few things to keep in mind while reviewing or testing in the future

- Make sure you do not have the app already installed or logged in. this will fail login tests
- When the suite runs through the login, the reading lists start downloading the articles which creates a lot of background work traffic and the espresso get deprioritized - doesn't crash but goes on a very long wait for a while till it times out. Make sure you use a new account or one with no saved pages
- If you end up using new account, you might have to make 3-4 edits on it so your SE tab is enabled so we can run the related tests [There is logic to check for any disable view, but if you want to run se tests you need this]

### CODE REVIEW

**Comment by sharvaniharan** [comment]
Marking this wip until we are able to run the test end to end in one go.

**Comment by sharvaniharan** [comment]
@dbrant @cooltey I removed the `wip` so we can merge the tests to keep things moving.

**Comment by cooltey** [comment]
Hi @sharvaniharan 

I did a couple of runs and still saw failed every time. Would it be possible to fix the issues before merging to the main branch? Thanks!

**Comment by sharvaniharan** [comment]
@cooltey @dbrant Please test this again. I was able to run end-to-end multiple times and have left a note in the pr notes as well.

**Comment by cooltey** [comment]
> @cooltey @dbrant Please test this again. I was able to run end-to-end multiple times and have left a note in the pr notes as well.

Tried a couple of times and it failed. Would you be able to fix it?

File: app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt [hunk 1]
```diff
@@ -516,15 +796,401 @@ class SmokeTests {
...
+
+        onView(allOf(TestUtil.withGrandparent(withId(R.id.login_password_input)), withClassName(Matchers.`is`("org.wikipedia.views.PlainPasteEditText"))))
+            .perform(replaceText(BuildConfig.TEST_LOGIN_PASSWORD), closeSoftKeyboard())
+
+        // Click the login button
+        onView(withId(R.id.login_button)).perform(scrollTo(), click())
+
+        TestUtil.delay(5)
+
+        onView(allOf(withId(android.R.id.button2), withText("No thanks"),
```
Comment (dbrant) [inline app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt:865]: It's all working for me, until this point. What/where is the "No thanks" button supposed to be? (what component is that part of?)

File: app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt [hunk 2]
```diff
@@ -516,15 +796,401 @@ class SmokeTests {
...
+
+        TestUtil.delay(2)
+
+        // Click the login button
+        onView(allOf(withId(R.id.create_account_login_button), withText("Log in"), isDisplayed()))
+            .perform(click())
+
+        TestUtil.delay(2)
+
+        // Type in an incorrect username and password
```
Comment (dbrant) [inline app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt:853]: It looks like all the following actions depend on a _correct_ username and password, which makes this comment confusing.

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt (hunk 1 above)
Comment (sharvaniharan) [reply app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt:865]: This is the lists sync dialog dismiss button :
image

It should appear right after login... will check again why it is failing 🤔

**Comment by sharvaniharan** [comment]
> 
> Tried a couple of times and it failed. Would you be able to fix it?


This is a very common one @cooltey 🙈  This happens when it tries to test explore feed cards by saving to list.. depending on the length of a card sometimes it is not visible and it fails... Scrolling to it when its not visible is flaky. I have a few options as to how to deal with it.. we can discuss on sync Monday and will update a decision here later since you are on vacation 🏖️

**Comment by sharvaniharan** [comment]
@dbrant I have added more checks for some login tests. Please test again and let me know.

**Pull request merged**